tests/describe_task_queue.go

```diff
@@ -277,3 +300,120 @@ func (s *DescribeTaskQueueSuite) validateDescribeTaskQueue(
+
+	// Poll the activity tasks
+	s.PollActivityTasks(workflows, tq, identity)
+
+	expectedBacklogCount[enumspb.TASK_QUEUE_TYPE_ACTIVITY] = int64(0)
+	expectedAddRate[enumspb.TASK_QUEUE_TYPE_ACTIVITY] = workflows > 0
+	expectedDispatchRate[enumspb.TASK_QUEUE_TYPE_ACTIVITY] = workflows > 0
+
+	s.validateDescribeTaskQueue(tqName, expectedBacklogCount, maxBacklogExtraTasks, expectedAddRate, expectedDispatchRate, isEnhancedMode)
+}
+
+// publishConsumeWorkflowTasksValidateStats method for the DescribeTaskQueueSuiteWithCache suite
+func (s *DescribeTaskQueueSuiteWithCache) publishConsumeWorkflowTasksValidateStats(workflows int, isEnhancedMode bool) {
+
+	tqName := s.randomizeStr("backlog-counter-task-queue")
+	tq := &taskqueuepb.TaskQueue{Name: tqName, Kind: enumspb.TASK_QUEUE_KIND_NORMAL}
+	identity := "worker-multiple-tasks"
+
+	expectedBacklogCount := make(map[enumspb.TaskQueueType]int64)
+	maxBacklogExtraTasks := make(map[enumspb.TaskQueueType]int64)
+	expectedAddRate := make(map[enumspb.TaskQueueType]bool)
+	expectedDispatchRate := make(map[enumspb.TaskQueueType]bool)
+
+	// Actual counter can be greater than the expected due to History->Matching retries. We make sure the counter is in
+	// range [expected, expected+maxExtraTasksAllowed]
+	maxExtraTasksAllowed := int64(3)
+	if workflows <= 0 {
+		maxExtraTasksAllowed = int64(0)
+	}
+
+	expectedBacklogCount[enumspb.TASK_QUEUE_TYPE_ACTIVITY] = 0
+	maxBacklogExtraTasks[enumspb.TASK_QUEUE_TYPE_ACTIVITY] = 0
+	expectedAddRate[enumspb.TASK_QUEUE_TYPE_ACTIVITY] = false
+	expectedDispatchRate[enumspb.TASK_QUEUE_TYPE_ACTIVITY] = false
+
+	s.StartWorkflowExecution(tqName, workflows)
+
+	expectedBacklogCount[enumspb.TASK_QUEUE_TYPE_WORKFLOW] = int64(workflows)
+	maxBacklogExtraTasks[enumspb.TASK_QUEUE_TYPE_WORKFLOW] = maxExtraTasksAllowed
+	expectedAddRate[enumspb.TASK_QUEUE_TYPE_WORKFLOW] = workflows > 0
+	expectedDispatchRate[enumspb.TASK_QUEUE_TYPE_WORKFLOW] = false
+
+	// cache has no entries, gRPC calls will be made to fetch the latest stats
+	s.validateDescribeTaskQueue(tqName, expectedBacklogCount, maxBacklogExtraTasks, expectedAddRate, expectedDispatchRate, isEnhancedMode)
+
+	// Poll and respond for the WF tasks
+	s.PollAndRespondWfTasks(workflows, tq, identity)
+
+	// call describeTaskQueue to verify if the WTF backlog decreased and activity backlog increased
+	expectedBacklogCount[enumspb.TASK_QUEUE_TYPE_WORKFLOW] = int64(0)
+	expectedAddRate[enumspb.TASK_QUEUE_TYPE_WORKFLOW] = workflows > 0
+	expectedDispatchRate[enumspb.TASK_QUEUE_TYPE_WORKFLOW] = workflows > 0
+
+	expectedBacklogCount[enumspb.TASK_QUEUE_TYPE_ACTIVITY] = int64(workflows)
+	maxBacklogExtraTasks[enumspb.TASK_QUEUE_TYPE_ACTIVITY] = maxExtraTasksAllowed
+	expectedAddRate[enumspb.TASK_QUEUE_TYPE_ACTIVITY] = workflows > 0
+	expectedDispatchRate[enumspb.TASK_QUEUE_TYPE_ACTIVITY] = false
+
+	time.Sleep(6 * time.Second) // forcing cache evictions due to TTL
```

Comment:
if you decrease the TTL to <5s, this sleep will be less of a big deal. The linter says "unless you've no other option" and I feel like testing a TTL is an obvious place to do a (short) sleep, so I'd support you ignoring the lint if you want, but you could also try pleasing the linter with something like:
```
	s.Eventually(func() bool {
		// fetches the latest stats by making per partition gRPC calls
		s.validateDescribeTaskQueue(tqName, expectedBacklogCount, maxBacklogExtraTasks, expectedAddRate, expectedDispatchRate, isEnhancedMode)
	}, testTTL+1, testTTL)
```